test/python/common.py

```diff
@@ -79,3 +79,23 @@ def _get_resource_path(filename, path=Path.TEST):
             str: the absolute path to the resource.
         """
         return os.path.normpath(os.path.join(path.value, filename))
+
+    def assertNoLogs(self, logger=None, level=None):
+        """The opposite to assertLogs.
+        """
+        return _AssertNoLogsContext(self, logger, level)
+
+class _AssertNoLogsContext(unittest.case._AssertLogsContext):
+    """A context manager used to implement TestCase.assertNoLogs()."""
+
+    LOGGING_FORMAT = "%(levelname)s:%(name)s:%(message)s"
+    def __exit__(self, exc_type, exc_value, tb):
+        self.logger.handlers = self.old_handlers
+        self.logger.propagate = self.old_propagate
+        self.logger.setLevel(self.old_level)
+        if exc_type is not None:
```

Comment:
This might not be picked by the linter - but probably it is clearer to do just `if exc_type`?

Reply:
This is a reimplementation  of `AssertLogsContext` and I prefer to stay as closer as possible to the original code: https://github.com/python/cpython/blob/3.6/Lib/unittest/case.py#L328
Except you have a strong opinion.